Source/Core/Core/Debugger/CodeTrace.h

```diff
@@ -31,26 +29,37 @@ struct InstructionAttributes
 
 struct TraceOutput
 {
-  u32 address = 0;
+  u32 address;
   std::optional<u32> memory_target = std::nullopt;
   std::string instruction;
+  std::array<double, 4> value{10, 10, 10, 10};
+  std::array<std::string, 4> regs{"", "", "", ""};
```

Comment:
Hmm the regs being "" like that is redundant. I should drop it.
The double - value, if I understand correctly, is initialized with some random value unless you write something in. I chose 10's as a debug value, but if you have any suggestions for a final value, that'd be great.

Size 4 is because there are at maximum 4 registers used at once.  Would I just do #DEFINE MAX_REGISTERS 4 at the top? Or use something else?

Reply:
Double is the maximum size  of a value.  Values can be mixed types like double + u32. "lfd" for example loads a double from a register's u32.  I'm not sure if variant works for that...?

Reply:
> The double - value, if I understand correctly, is initialized with some random value unless you write something in. I chose 10's as a debug value, but if you have any suggestions for a final value, that'd be great.

Assuming that value is used to store the register content (if matching the regex), you could use a `std::optional<double>` instead so it has a double value or no value at all (if no match).



> Size 4 is because there are at maximum 4 registers used at once. Would I just do #DEFINE MAX_REGISTERS 4 at the top? Or use something else?

If I understand correctly, it's the maximum number of operands in the instruction (that can be matched by the regex), so you can define a constexpr variable like `constexpr int MAX_OPERANDS` or `MAX_USED_REGISTER`, to make it more obvious. Moreover, there are, at several places, `for` loops based on `<= 3` which is hardcoded and doesn't depend on the size of your container. So it if changes for whatever reason it might introduce issues and make the code less flexible.

> I may need to change the way this works. The value of each double must correspond to the regs with the same index and the first reg has to have its double value filled last (because we have to step before filling it). A bit tricky.

Have you tried `std::map` or a vector/array of `std::pair<std::string,double>` to keep track of the (double) value associated with its register rather than using two distinct arrays?

Reply:
Alright, I thought hard and came up with a simpler implementation. Put it in its own commit.